s3file/static/s3file/js/s3file.js

```diff
@@ -80,15 +80,13 @@
   }
 
   function clickSubmit (e) {
-    e.preventDefault()
     let submitButton = e.target
     let form = submitButton.closest('form')
     const submitInput = document.createElement('input')
     submitInput.type = 'hidden'
     submitInput.value = submitButton.value || true
     submitInput.name = submitButton.name
     form.appendChild(submitInput)
-    uploadS3Inputs(form)
```

Comment:
@tilboerner how does this still work?

Reply:
It still works because with the other removed line in this patch, the default action for the click event is *not* prevented. That default action is to cause a 'submit' event on the form. (Note that the function containing the change is not handling the submit event itself, but the click event leading up to it.)

The submit event is then handled by the `form.onsubmit`, which was registered below, on 'DOMContentLoaded': 

https://github.com/codingjoe/django-s3file/blob/7e7656d31ccef881c9f31a3c54091bbba7631085/s3file/static/s3file/js/s3file.js#L111-L114

If I understand JS correctly, things proceed synchronously, and we can trust at this point that if the submit was triggered by a click, the click handler has added the hidden input to the form.

I found one caveat: When manually dispatching a click event for the button, it needs to be a `MouseEvent` to trigger the submit logic; a generic `Event` triggers the click handler, but does not default to submitting the form. Without the change in this PR, any old `Event` would do. 

However, the canonical [click event *is* a MouseEvent][], and gets dispatched as such even when triggering the button by keyboard or calling `button.click()`. All things considered, I think it's better to make the proposed change and honor the expectation of other submit handlers that they will be called on submit, rather than cater to code that uses non-standard, non-`MouseEvents` to trigger submit clicks.

[click event *is* a MouseEvent]: https://developer.mozilla.org/en-US/docs/Web/Events/click